tests/cli/test_integration_datasets.py

```diff
@@ -1293,14 +1293,56 @@ def test_renku_clone_with_config(tmpdir):
             }
         )
 
-        repo = git.Repo('datasets-test')
+        assert 'master' == str(repo.active_branch)
         reader = repo.config_reader()
         reader.values()
 
         lfs_config = dict(reader.items('filter.lfs'))
         assert '0' == lfs_config.get('custom')
 
 
+@pytest.mark.integration
+@flaky(max_runs=10, min_passes=1)
+def test_renku_clone_checkout_rev(tmpdir):
+    """Test cloning of a Renku repo checking out a rev."""
+    remote = 'https://dev.renku.ch/gitlab/virginiafriedrich/datasets-test.git'
+
+    with chdir(str(tmpdir)):
+        repo = project_clone(
+            remote,
+            config={
+                'user.name': 'sam',
+                'user.email': 's@m.i',
+                'filter.lfs.custom': '0'
+            },
+            checkout_rev='3d387e64ea25079df8dd43b8875058cf9f4b0315',
+        )
+
+        assert '3d387e64ea25079df8dd43b8875058cf9f4b0315' == str(
+            repo.active_branch
+        )
+        reader = repo.config_reader()
+        reader.values()
+
+        lfs_config = dict(reader.items('filter.lfs'))
+        assert '0' == lfs_config.get('custom')
```

Comment:
Are these last 4 statements required for this test?

Reply:
I would like to keep it and reason being is because this is exact case how API is being used from the service perspective so having an additional check for specific ref with custom git config gives us a nice safety net for that case.